hazelcast-jet-core/src/main/java/com/hazelcast/jet/windowing/WindowingProcessors.java

```diff
@@ -88,14 +89,11 @@ private WindowingProcessors() {
      * output.
      *
      * @param <K> type of the grouping key
-     * @param <F> type of the frame
+     * @param <F> type of accumulator
      * @param <R> type of the result derived from a frame
-     * @param emitPunctuation If {@code true}, punctuation will be emitted when window
-     *                        is closed. Enable, if downstream vertex needs to know,
-     *                        that it has received all keys for a particular window.
      */
     public static <K, F, R> Distributed.Supplier<SlidingWindowP<K, F, R>> slidingWindow(
-            WindowDefinition windowDef, WindowOperation<K, F, R> windowOperation, boolean emitPunctuation) {
+            WindowDefinition windowDef, WindowOperation<?, F, R> windowOperation) {
```

Comment:
Seems like we have no basis to infer `K` here, it can resolve to anything. This is a part of the wider concern on what to declare as the return type. We could have just `Distributed.Supplier<Processor>` if we don't want to expose the implementation type, but then we lose all the type information; OTOH the type information is arbitrary and could be wrong.